### PR TITLE
Added informal advisory to mz-avro

### PR DESCRIPTION
**Pull request by jorgecarleitao**

### CODE REVIEW

**Comment by pinkforest** [comment]
@amousset would like to hear your opinion on this one? 

https://github.com/MaterializeInc/materialize/issues/8669
https://github.com/MaterializeInc/materialize/pull/9944
https://github.com/MaterializeInc/materialize/pull/9983

Seems the disclosure was done appropriately, the maintainer reacted acknowledging the issue and the issue was fixed in the repository however no crate was uploaded to crates.io that would contain fix - the crate version remains not yanked.

There is only 272 downloads on this crate - I think this crate may not be supposed to be used via crates.io ?

**Comment by pinkforest** [comment]
Okay the fix is incoming -
https://github.com/MaterializeInc/materialize/pull/14087

We can set the patched as 0.7.0 - waiting on it getting published - the old version has been yanked

**Comment by pinkforest** [comment]
Hi @jorgecarleitao the patched version is not in crates.io - 
https://github.com/MaterializeInc/materialize/pull/14087#issuecomment-1212825554

Could we please update as discussed above and we should be able to merge - thanks

**Comment by jorgecarleitao** [comment]
Added patched

**Review by amousset — COMMENTED** [review]
LGTM, just needs a small change on patched version

**Comment by pinkforest** [comment]
@jorgecarleitao - Also please address the project maintainer comment: https://github.com/MaterializeInc/materialize/issues/8669#issuecomment-1205723449

`Sure, but please point out in the advisory that there is only UB in the case where a user provides a struct whose Read impl inspects its read_exact buffer before writing into it. It's indeed possible to use the crate unsoundly, but only by writing a highly unidiomatic Read impl.`

Also the date still needs to be adjusted as per my original review comments - thanks a lot :)

**Review by pinkforest — COMMENTED** [review]
@jorgecarleitao - Also could we please address the project maintainer comment: https://github.com/MaterializeInc/materialize/issues/8669#issuecomment-1205723449

`Sure, but please point out in the advisory that there is only UB in the case where a user provides a struct whose Read impl inspects its read_exact buffer before writing into it. It's indeed possible to use the crate unsoundly, but only by writing a highly unidiomatic Read impl.`

Also the date still needs to be adjusted as per my original review comments - thanks a lot :)

File: crates/mz-avro/RUSTSEC-0000-0000.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+date = "2022-01-08"
+url = "https://github.com/MaterializeInc/materialize/issues/8669"
+categories = ["memory-exposure"]
+informational = "unsound"
+
+[versions]
+patched = []
+```
+
+# Incorrect use of `set_len` allows for un-initialized memory
```
[the remark this answers is not in the document]
Comment (jorgecarleitao) [reply crates/mz-avro/RUSTSEC-0000-0000.md:14]: I have added such note (with minor rewording)

[the remark this answers is not in the document]
Comment (pinkforest) [reply crates/mz-avro/RUSTSEC-0000-0000.md:14]: thanks a lot :)

**Comment by pinkforest** [comment]
looks good - thanks for the patience on this

**Pull request merged**